src/qt/frame.cpp

```diff
@@ -53,7 +53,8 @@ wxFrame::~wxFrame()
 bool wxFrame::Create( wxWindow *parent, wxWindowID id, const wxString& title,
     const wxPoint& pos, const wxSize& size, long style, const wxString& name )
 {
-    m_qtWindow = new wxQtMainWindow( parent, this );
+    wxWindow *effectiveParent = style & wxFRAME_NO_TASKBAR ? parent : NULL;
+    m_qtWindow = new wxQtMainWindow(effectiveParent, this );
```

Comment:
This breaks window management.

Before these changes, if you had 2 frames, one being the child of the other, when closing the parent frame, the child would also be destroyed and the app would exit. With the changes here, after closing the parent frame, the child will still remain on screen and the app will continue to run.

Reply:
Have you got any suggestions on how to work around this?  Without this change child windows don't behave very well.

Reply:
What is exactly the problem with the child frames right now? I understand that `wxFRAME_NO_TASKBAR` is not supported, but what else?

You might find the code/comments in `wxTopLevelWindowMSW::MSWGetStyle()` useful... perhaps we can do something similar with Qt?

Reply:
Thanks of the info about the MSW port,  I'll have a look at that

The wxFrame styles concerning the system menu, minimise and maximise buttons are also affected.

Qt prevents wx from removing the system menu, minimise and maximise buttons for secondary windows.  It might be possible to work round this by mapping the frame styles to a widget type and setting the type flag for the window.